ext/opcache/zend_file_cache.c

```diff
@@ -1813,102 +1819,58 @@ static void zend_file_cache_unserialize(zend_persistent_script  *script,
 	zend_file_cache_unserialize_early_bindings(script, buf);
 }
 
-zend_persistent_script *zend_file_cache_script_load(zend_file_handle *file_handle)
+zend_result zend_file_cache_validate_and_open(zend_file_handle *file_handle, zend_file_cache_handle *cache_handle)
```

Comment:
Instead of all these changes below, I would just add ``bool validate_only`` argument to ``zend_file_cache_script_load()`` and return true after header validation and before arena allocation and actual script loading.

Reply:
Thank you for the suggestion @dstogov !

I've made the modifications to do it this way, however, since `zend_file_cache_script_load()` returns `zend_persistent_script *`, I couldn't return `true` - I've had to mock that by returning a fixed, static instance of the `zend_persistent_script` struct.

Does this fit your expectations?